frontend/apps/reader/readerui.lua

```diff
@@ -329,6 +330,7 @@ end
 
 function ReaderUI:showReader(file)
     DEBUG("show reader ui")
+    ReadHistory:addItem(file)
```

Comment:
Since we are using O(n * log(n)) for sort, O(1) for delete, and O(n) for read. 1000 and 500 won't make much difference. I agree 500 books are out of track.
P.S. Until we finally removed history folder, we still need to read over 500 items, but only reduce after sort.